eng/Versions.props

```diff
@@ -3,6 +3,7 @@
 <Project>
   <PropertyGroup>
     <VersionPrefix>17.5.0</VersionPrefix>
+    <PackageValidationBaselineVersion>17.3.2</PackageValidationBaselineVersion>
```

Comment:
Ideally, bumping this version should be part of rebranding the repo to the next version. Do you have documentation that should be updated?

Reply:
We generally rebrand `main` before the public push of the release branch. So I think this'll need to be two-phase: at rebrand time, set the baseline to the current prerelease package, then after release update the baseline to the final public release.

Added that info to #8246 (we'll automate the creation and update of these checklists Someday™️)